src/components/AllComponents.js

```diff
@@ -0,0 +1,539 @@
+
+import { useQuery, useLazyQuery } from "@apollo/client";
+import { QUERY_CHARACTERS, QUERY_ONECHARACTER, GET_USERDATA, GET_USERCHARACTERSBYID } from "../util/queries";
+import CardDetails from "./CardDetails";
+
+import Auth from "../util/auth";
+
+function AllComponents() {
+  const categorySelect = document.getElementById("categories");
+
+  const [search, setSearch] = useState("");
+  const [characters, setCharacters] = useState([]);
+  const [characterCategory, setCharacterCategory] = useState("");
+
+  const [rarityCategory, setRarityCategory] = useState("ALL");
+  const [filteredCharacters, setFilteredCharacters] = useState([]);
+  const [typeCharacters, setTypeCharacters] = useState("ALL");
+  const [cardDetails, setCardDetails] = useState({
+    id: 1331,
+    thumb: 1003310,
+    art: null,
+    name: "Buu (Kid)",
+    category: ['Resurrected Warriors', 'Majin Buu Saga', 'Transformation Boost', 'Artificial Life Forms', 'Majin Power', 'Planetary Destruction', 'Storied Figures', 'Legendary Existence', 'Sworn Enemies', 'Accelerated Battle', 'Worldwide Chaos', 'Battle of Fate'],
+    link_skill: ['Majin', 'Brutal Beatdown', 'More Than Meets the Eye', 'Big Bad Bosses', 'Infinite Regeneration', 'Fierce Battle', 'The Wall Standing Tall'],
+    type: "EPHY",
+    rarity: "UR",
+    ls_description: `Allies’ ATK increases (MAX +50%) based on HP left`,
+    sa_name: "Planet Burst",
+    sa_description: "Causes immense damage to enemy and lowers DEF  <Lowers enemy's DEF by 40% for 3 turns>  ",
+  });
+  const [suggestion, setSuggestion] = useState([]);
+  const [webOfTeam, setWebOfTeam] = useState([]);
+
+  const [urActive, setUrActive] = useState(false);
+  const [lrActive, setLrActive] = useState(false);
+
+  const [aglActive, setAglActive] = useState(false);
+  const [teqActive, setTeqActive] = useState(false);
+  const [intActive, setIntActive] = useState(false);
+  const [strActive, setStrActive] = useState(false);
+  const [phyActive, setPhyActive] = useState(false);
+  const [allTypeActive, setAllTypeActive] = useState(true);
+
+  
+  const { loading, data } = useQuery(QUERY_CHARACTERS);
+  const allCharacters = data?.characters || [];
+  
+  const [getUserData, { loading:loading3, data:data3 }] = useLazyQuery(GET_USERDATA);
+  const [getUserCharactersById, { loading:loading4, data:data4 }] = useLazyQuery(GET_USERCHARACTERSBYID);
+
+  const [onLoadGetUserCharacters, setOnLoadGetUserCharacters] = useState([]);
+  const [userCharacters, setUserCharacters] = useState([]);
+  
+  useEffect(() => {
+    console.log('Not sure if logged in')
+
+    if(Auth.loggedIn()) {
+      console.log("logged in")
+      const username = Auth.getProfile().data.username
+      console.log(username);
+      getUserData({
+        variables: {
+          username: username
+        },
+      }).then((result) => {
+        setOnLoadGetUserCharacters(result.data.me.savedCharacters)
+        console.log(result);
+      });
+    }
+  },[])
+
+  useEffect(() => {
```

Comment:
You can do this as an API call, but you also have allCharacters already loaded from another API call. So you could skip the userCharacters state and just create
```
const userCharacters = allCharacters.filter(character => userCharacterIds.includes(character.id));
```

Alternatively, if you do want to do this as an API call, I'd try to rewrite the API call to combine it with the `getUserData` query to improve performance, i.e. rewrite the serverside resolver to make the savedCharacters be the list of character objects and not just a list of ids (though you'll probably want to set that to a new name, not make a breaking change). Benefits of doing it as an API call is that if you change the allCharacters call (e.g. make it paginated if there are too many characters) then this still works fine, downside is the larger work to rewrite the resolver. 

IMO I think it's fine to just filter allCharacters and skip the getById API call for now